lib/jsonmerger.js

```diff
@@ -53,7 +54,13 @@ module.exports = function jsonMerger(files, options) {
 
       // return false when hostname is matched; short-circuiting further
       // iteration
-      return !_.includes(stage.hosts || [], hostname);
+      if (!_.isUndefined(stage.env) && stage.env === process.env[options.stageEnv]) {
+        return false;
+      }
+      if (_.includes(stage.hosts || [], hostname)) {
+        return false;
+      }
+      return true;
```

Comment:
You could shorten this by using `return (negated stageEnv-test) && (negated host-test)`